dttools/src/hash_table.h

```diff
@@ -170,7 +170,9 @@ HASH_TABLE_ITERATE(table,key,value) {
 </pre>
 */
 
-#define HASH_TABLE_ITERATE( table, key, value ) hash_table_firstkey(table); while(hash_table_nextkey(table,&key,(void**)&value))
+extern int (*hash_table_half_buckets)(struct hash_table *);
```

Comment:
rather than halving here, let's do it inside `hash_table_firstkey` and `hash_table_randomkey`.